mmv1/third_party/tgc_next/tfplan2cai/models/fake_resource_data.go

```diff
@@ -0,0 +1,358 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// In order to interact with resource converters, we need to be able to create
+// "terraform resource data" that supports a very limited subset of the API actually
+// used during the conversion process.
+package models
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Must be set to the same value as the internal typeObject const
+const typeObject schema.ValueType = 8
+
+// This is more or less equivalent to the internal getResult struct
+// used by schema.ResourceData
+type getResult struct {
+	Value    interface{}
+	Computed bool
+	Exists   bool
+	Schema   *schema.Schema
+}
+
+// Compare to https://github.com/hashicorp/terraform-plugin-sdk/blob/97b4465/helper/schema/resource_data.go#L15
+type FakeResourceData struct {
+	reader     schema.FieldReader
+	kind       string
+	tfplanAddr string
+	isDeleted  bool
+	schema     map[string]*schema.Schema
+}
+
+// Kind returns the type of resource (i.e. "google_storage_bucket").
+func (d *FakeResourceData) Kind() string {
+	return d.kind
+}
+
+// Id returns the ID of the resource from state.
+func (d *FakeResourceData) Id() string {
+	return ""
+}
+
+func (d *FakeResourceData) TfplanAddr() string {
```

Comment:
This function is new.

Reply:
Rather than TfplanAddr for this & the field name, I'd suggest just "Address" since that's the name used in Terraform generally: https://developer.hashicorp.com/terraform/cli/state/resource-addressing.